docs/reference/aggregations/metrics/top-metrics-aggregation.asciidoc

```diff
@@ -0,0 +1,281 @@
+come here.
+
+==== Examples
+
+===== Use with terms
+
+This aggregation should be quite useful inside of <<search-aggregations-bucket-terms-aggregation, `terms`>>
+aggregation, to, say, find the last value reported by each server.
+
+[source,console,id=search-aggregations-metrics-top-metrics-terms]
+----
+PUT /node
+{
+  "mappings": {
+    "properties": {
+      "ip": {"type": "ip"},
+      "date": {"type": "date"}
+    }
+  }
+}
+POST /node/_bulk?refresh
+{"index": {}}
+{"ip": "192.168.0.1", "date": "2020-01-01T01:01:01", "v": 1}
+{"index": {}}
+{"ip": "192.168.0.1", "date": "2020-01-01T02:01:01", "v": 2}
+{"index": {}}
+{"ip": "192.168.0.2", "date": "2020-01-01T02:01:01", "v": 3}
+POST /node/_search?filter_path=aggregations
+{
+  "aggs": {
+    "ip": {
+      "terms": {
+        "field": "ip"
+      },
+      "aggs": {
+        "tm": {
+          "top_metrics": {
+            "metric": {"field": "v"},
+            "sort": {"date": "desc"}
+          }
+        }
+      }
+    }
+  }
+}
+----
+
+Which returns:
+
+[source,js]
+----
+{
+  "aggregations": {
+    "ip": {
+      "buckets": [
+        {
+          "key": "192.168.0.1",
+          "doc_count": 2,
+          "tm": {
+            "top": [ {"sort": ["2020-01-01T02:01:01.000Z"], "metrics": {"v": 2.0 } } ]
+          }
+        },
+        {
+          "key": "192.168.0.2",
+          "doc_count": 1,
+          "tm": {
+            "top": [ {"sort": ["2020-01-01T02:01:01.000Z"], "metrics": {"v": 3.0 } } ]
+          }
+        }
+      ],
+      "doc_count_error_upper_bound": 0,
+      "sum_other_doc_count": 0
+    }
+  }
+}
+----
+// TESTRESPONSE
+
+Unlike `top_hits`, you can sort buckets by the results of this metric:
+
+[source,console]
+----
+POST /node/_search?filter_path=aggregations
+{
+  "aggs": {
+    "ip": {
+      "terms": {
+        "field": "ip",
+        "order": {"tm.v": "desc"}
+      },
+      "aggs": {
+        "tm": {
+          "top_metrics": {
+            "metric": {"field": "v"},
+            "sort": {"date": "desc"}
+          }
+        }
+      }
+    }
+  }
+}
+----
+// TEST[continued]
+
+Which returns:
+
+[source,js]
+----
+{
+  "aggregations": {
+    "ip": {
+      "buckets": [
+        {
+          "key": "192.168.0.2",
+          "doc_count": 1,
+          "tm": {
+            "top": [ {"sort": ["2020-01-01T02:01:01.000Z"], "metrics": {"v": 3.0 } } ]
+          }
+        },
+        {
+          "key": "192.168.0.1",
+          "doc_count": 2,
+          "tm": {
+            "top": [ {"sort": ["2020-01-01T02:01:01.000Z"], "metrics": {"v": 2.0 } } ]
+          }
+        }
+      ],
+      "doc_count_error_upper_bound": 0,
+      "sum_other_doc_count": 0
+    }
+  }
+}
+----
+// TESTRESPONSE
+
+===== Mixed sort types
+
+Sorting `top_metrics` by a field that has different types across different
+indices producs somewhat suprising results: floating point fields are
+always sorted *before* whole number fields.
```

Comment:
Should we mention this is an implementation detail so that we feel free to change this behavior in the future?

Reply:
++ - or maybe something a little more vague.